### ISSUE TITLE
`canGetProperty()` and `canSetProperty()` does not take in account `ActiveRecord::attributes()`

### ISSUE THREAD

**Issue by klimov-paul** [yiisoft/yii2#9561]
Methods `Object::canGetProperty()` and `Object::canSetProperty()` does not take in account `ActiveRecord::attributes()`, always returning `false` for them:

```php
/**
 * Properties, which are attributes from source table:
 *
 * @property integer $id
 * @property string $username
 */
class User extends \yii\db\ActiveRecord
{
    ...
}

$user = new User();
$this->assertTrue($user->canGetProperty('id')); // FAILS !!!
$this->assertTrue($user->canSetProperty('id')); // FAILS !!!
```

Inside `yii\db\BaseActiveRecord` we override `__set()` and `__get()` methods, but there is no redeclaration of `canGetProperty()` or `canSetProperty()`.

**Comment by ravshanium** [comment]
BaseActiveRcord extends Model, Model extends Component, Component extends Object but functions canGetProperty() and canSetProperty are redeclared in Component (behaviors added). Don't you think it is better to redeclare the function in BaseActiveRecord within adding block of searching key in attributes, do you? 
Another question why in Component you don't use parent::canGetProperty, but copy the code from Object?

**Comment by dynasource** [comment]
if a attribute is seen as a 'settable' property, which I do think is logical and in fact rather straightforward, I would assume that both would include a check on BaseActiveRecord::hasAttribute().

**Comment by klimov-paul** [comment]
Resolved by commit 75439d367d2d1270bb61642f236df6a552acf29e
